ocaml/xapi/recommendations.mli

```diff
@@ -0,0 +1,15 @@
+(*
+ * Copyright (c) Cloud Software Group, Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+val load : string -> (string * string) list
```

Comment:
There should be some documentation here what this does. The parameter could be a label. 
```
val load: dir:string -> (string * string) list
```
I think returning a StringMap.t would prevent inefficient list operations by clients. This would require exposing the StringMap:

```
module StringMap: Map.S with type key = string

val load: dir:file -> string StringMap.t 
```

Reply:
What means `dir:file` here? 
file -> string?

Reply:
I am suggesting to use a labeled argument (could be label `path` as well) to assign some meaning to this string in the interface.